lab-tara/image-app/app/src/main/java/com/example/imageapp/MainActivity.java

```diff
@@ -0,0 +1,117 @@
+            new ClevelandSportsCurse("The Move", R.drawable.the_move,
+                    "More than a decade has passed since \"The Move,\" but the name Art Modell still draws angry looks and responses when mentioned within Cleveland's city limits. His decision to move the storied Browns franchise to Baltimore, Md., for the start of the 1996 NFL season led to angry protests, lawsuits and ultimately the NFL allowing the team to keep its name, team colors and history intact for a new incarnation of the franchise three seasons later."),
+            new ClevelandSportsCurse("1997 World Series", R.drawable.world_series,
+                    "The Marlins tied Game 7 against Matt Williams (below) and the Indians in the bottom of the ninth, then went on to claim the game and series in the 11th. More recently, the Indians blew a 3-1 lead to the Boston Red Sox in the 2007 ALCS and still have not returned to the World Series since 1997. Cleveland has the second-longest World Series championship drought (1948) behind the Chicago Cubs."),
+            new ClevelandSportsCurse("The Decision", R.drawable.the_decision,
+                    "\"This fall I am taking my talents to South Beach and play with the Miami Heat. The major factor was the best opportunity for me to win, to win now and for the future also. Winning is the most important thing for me. I feel like this is going to be the best opportunity.\" — LeBron James, July 8, 2010")
+    };
+
+    private TextView page;
+    private TextView name;
+    private ImageView picture;
+    private TextView description;
+    private Button previous;
+    private Button next;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        currentIndex = 0;
+
+        page = findViewById(R.id.page);
+        name = findViewById(R.id.name);
+        picture = findViewById(R.id.picture);
+        description = findViewById(R.id.description);
+
+        previous = findViewById(R.id.previous);
+        next = findViewById(R.id.next);
+
+        showPage();
+
+        showCurse(currentIndex);
+
+        previous.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                decrement();
+                showPage();
+            }
+        });
+
+        next.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                increment();
+                showPage();
+            }
+        });
+    }
+
```

Comment:
The below methods all feel like implementation details that might be best made private. When you are trying to decide what to make public/protected/private, the inclination should be to start as restrictive as possible and only expose what absolutely needs to be externally accessible.